### ISSUE TITLE
`else` followed by `//comment` and then `if` has incorrect indention

### ISSUE THREAD

**Issue by nhirschey** [fsprojects/fantomas#2449]
Issue created from [fantomas-online](https://fsprojects.github.io/fantomas-tools/#/fantomas/preview?data=N4KABGBEDGD2AmBTSAuKBLAZmALgJwFdFcALRAOwB09LyIIAHPdcnTOyyAQ0%2BtsQA2AZ2IB6UaXRCwUsFzBwAtooo4%2BdelnX0dYTF2GJt9HGSo0NusExZsOkAEa8LOwSONWbrdmE7RntOpedr6Q8JyQADTgUCI4OCwA5kKoYADaMRCgVpAAJDgAngzIaJC2UZn0eQBuBkSpGVYQAAzRTVAAkuRIrADK6ABeyG1NACyVEAC6lQC%2BI1kTUPlFJRisFTm5tQL1aI1NAIzzupAAslwAHgAyLIhXFImmG4cATM2L0zpzldkny8WpSAUeAAeUwN3IiF6hQEw0WNTqq32Vhexx0kAAot0wRC4e1IAJMJAPrNjr90f9Vo5YLABM8KdtdulFhAAMxoqpdER4HAAMRYBgAcogAO4CW703T4Igkr5k%2BGUwEOGl0jkInZIllgUYciCQXoMLjQRAAIUQmFgeEQAAUuHguCocIg8JKdNLELL6N8dOSqorSsraa6lozNe0AKy6qAGo2m82Wu6wEXO6BcERdaqwVMJWDkYMQd2eiDe%2Bi%2BvX%2BqCB1UK0MNLUANij%2BsNxrNFqtAFUGMU8Kn0%2BRM9n0Ln83oDCIi2ASwtNoUAQGVcH1UzkboAOxNmOt%2BNWgDCAjTQl3uaE0ugOEto-0hkn07AZaWc6pVaXW0RdfaAA5Ny24%2B3EKciCKA4zpXuOHpWJ8XryrOKxKouapvhqH5NAAnD%2BsZtgmx4CCOUbXhOkGkj8CpPvBQaIbWexagcrRas2sYAIKYE6eDHsoXCjoWRFyiRsHzpWCE1u%2B1HtAcRz0Vuf4Jr0QHoHAuF5vh4G3jBfxkQuFHCchomvBhxrMaxsmKPJtJ4Vq3G6FBxZqRSGmCVpmxUcyYnspJv6MXgsAEN0AAigjoCZrFcYQEFWcRPqkXBpTlJRIkuYcOr0ecFwdJgAAqZi9CQlo4AA6ug8BPFG7w8dBfHqdFaw4K%2BzmrjoByRsllxpZlFAYoY2W5QVRUkKO9aleFvGRfxVKxdpK40Y2zWpeQmDoBcII9lwF54BiFxMIgQhCMOSlap%2Bg06NZU62X69llOscU6QlVgHBuM0AEqIHAeDwD1xVaqMh30Mdd4Pnk53jU58X1fQBzfo9z2WvAgoEMBzpgh0TqKCkUYHKpFV2VVkCKAQAgJOKkK8paigrax6VwVdk1ieh9FPS98CnHjBO3MTeCk-EoFNtAJB2karEAPoioVTwYyNlUCRdNVU2GTRvE2KWefaBQgngNynu9fVRgdYullFktA38dVai8En4oreDK6r6s4LD8N4IjyOozRuszhLVK4-j6CE4gbMc%2BTlMTbLKKovRStcCratSDgTNez7ftk1z9E83z57OkLIt9a79762Nl1ByhKJueblwAGqIiaLDwEkmujjrZU2ZjZ3Y4bDIgybSUlxcvI%2Beeu2V90NeZ6OX3Z-9FZS7V7ftC8TVd75sA4AA4ogODrZt227bX2vfVMEV66N5HVsD12gxALzTebzPe7cJq4dAADWJr2o-q9CCC5C9A6iA4XDe3tARMKR195uyxpLF8MtC66BePdfEwoxS31XsmCgFNij%2BXmuQdAOZyCMW6IBe2zt2iWWAcNA%2B7sj5T1PibCG%2BJGLikSOQHu5A%2B65n6AwlaBArTpVgF0HoOAVq7TAjeBuJ0m7lnOhAguukUS01ofjZ05ABHVETOQRI%2BCQJ4HQSwLBu1CFNEAWPXOFDIHSN0KyOiV8va4iuA6Bw8AuD7lgDtVR8CfZCMIkNcq4swHPiEifamTRWRmyaJiDazpAqqAMAAaUQIgBgvDVBdGflwZhWstQGJEX9Ixmlj5G2noE0O%2BI74pIfriIQnlvLdGFKeRAjNr4%2B3XlaTeJ4QoykyadcR2NJH%2BODmY4uISTR2iwlaXyUhoDMBMoop08AOyYNzP5aAB57TYPcUAn6ICc6HxyZQgJVhWSdxCevcJKhWAGGhF5Agp5CAMGhAUWEqzDFbOqjs3pOhWRzxCTEuJKU7YaLBMU8gpTbh6Nus0XeYBfodMfF0vxeSqHtFZJfEJ5z5IxwQHifRKlMkxEmCMMoQheQ7VSAYmYQA)

#### Code

```fsharp
if true then
    printfn "a"
else // this is a comment
    if
        false
    then
        printfn "b"
    else
        printfn "c"

printfn "d"
```
            

#### Result

```fsharp
if true then
    printfn "a"
else // this is a comment
    if false then printfn "b" else printfn "c"

    printfn "d"

```

#### Problem description

*Note: The above auto-generated result is not the actual code from the webpage or in my source code using "fantomas version": "5.0.0-beta-007".* but it is wrong too.

Actual formatted result using the fantomas webtool:

```fsharp
if true then
    printfn "a"
else // this is a comment
if
    false
then
    printfn "b"
else
    printfn "c"

printfn "d"
```

Original code output
```fsharp
> if true then
-     printfn "a"
- else // this is a comment
-     if
-         false
-     then
-         printfn "b"
-     else
-         printfn "c"
-
- printfn "d";;
a
d
val it: unit = ()

```

Formatted code output:
```fsharp
> if true then
-     printfn "a"
- else // this is a comment
- if
-     false
- then
-     printfn "b"
- else
-     printfn "c"
-
- printfn "d";;
a
val it: unit = ()
```
            
**Another related incorrect format, putting the comment on its own line**

code:
```fsharp
if true then
    printfn "a"
else
    // comment
    if true then
        printfn "b"
    else
        printfn "c"

printfn "d"
```
result
```fsharp
if true then
    printfn "a"
else
// comment
if
    true
then
    printfn "b"
else
    printfn "c"

printfn "d"
```

#### Extra information

- [x] The formatted result breaks my code.
- [ ] The formatted result gives compiler warnings.
- [ ] I or my company would be willing to help fix this.

#### Options

Fantomas master branch at 2022-08-24T08:26:36Z - bc406e217b72e6ee262fb7575e401eee109cf799

Default Fantomas configuration

<sub>Did you know that you can ignore files when formatting from fantomas-tool or the FAKE targets by using a [.fantomasignore file](https://github.com/fsprojects/fantomas/blob/master/docs/Documentation.md#ignore-files-fantomasignore)?</sub>

**Comment by nojaf** [comment]
Hello, this is an idempotency issue, in the future, please report it as such using the other button:

![image](https://user-images.githubusercontent.com/2621499/187160188-98eef442-28d8-400e-aee3-b896f1e4fd29.png)

The initial formatting is correct according to the style guide:

```fsharp
if true then
    printfn "a"
else // this is a comment
if
    false
then
    printfn "b"
else
    printfn "c"

printfn "d"
```

The `if` expression doesn't fit on one line so the 
```fsharp
elif
    condExpr
then
    branchExpr
```

kicks in.

However, due to the comment, the semantics changed.
The `if/ else if/ else` changed to a nested `if / else` inside an `if / else`.
I'm not quite sure this is worth fixing, as putting the comment pretty much anywhere else resolves the problem. 

I would accept a PR only if the fix is clean and isolated.
Are you interested in submitting such a PR?

**Comment by nhirschey** [comment]
Thanks for taking the time to look at this.

> this is an idempotency issue

Just to make sure this was clear: I filed the issue because *fantomas broke my code*. Fantomas produced code that compiled and that followed the style guide, but my unit tests caught that fantomas changed how the code functioned.

Specifically, in my simplified examples the output produced by my original code is different from the output produced by either fantomas-formatted version. 

The idempotency problem is separate and only materialized after I used the web tool to create this issue. Sorry for confusing things.

> Are you interested in submitting such as PR?

I have zero experience with the AST and I am not confident that I could help. In time I could try to take a stab at it, but it's not an easy fix for me.

> putting the comment pretty much anywhere else resolves the problem.

For anybody who hits this, there are 3 places where I see putting this comment in my example, 2 of them result in broken code after using fantomas. One is safe with fantomas :
```fsharp
// Formatting this with fantomas breaks code
if true then
    printfn "a"
else // this is a comment
.
.
.
```

```fsharp
// Formatting this with fantomas breaks code
if true then
    printfn "a"
else 
    // this is a comment
    if 
.
.
.
```


```fsharp
// Formatting does not break this code
if true then
    printfn "a"
// this is a comment
else 
    if 
.
.
.
```

**Comment by nojaf** [comment]
I see now that your code is broken, but having the comment after the `else` keyword just messes up everything. I'm not even sure we can fix this.
The comment makes the entire if expression multiline, leading to

```fsharp
elif 
    a
else
    b
```
Both when it is after the `else` or before the `if`. Which changes how the parser will look at the code.

**Comment by nojaf** [comment]
Hmm, perhaps it is an option to restore in both cases the trivia after the `if` keyword.
Regardless of the original assignment.
We just need to write `else if` in one piece, there is no wiggle room I fear.

**Comment by nojaf** [comment]
Fix available in [v5.2.0-beta-001](https://www.nuget.org/packages/fantomas/5.2.0-beta-001)

**Comment by nhirschey** [comment]
Many thanks @nojaf!
